### PR TITLE
DRY up crafter/apprentice duplicate logic

### PR DESCRIPTION
**Pull request by laurentqro**

### CODE REVIEW

File: lib/commands/crafter/add_office.rb [hunk 1]
```diff
@@ -21,41 +21,46 @@ def prepare(data)
       @apprentice = Apprentice.profile(data[:user_id])
     end
 
-    def parse_office_location(data)
-      data[:user_message].gsub("office:", "").strip.downcase
+    def run
+      return ""                 unless user
```
Comment (fabientownsend) [inline lib/commands/crafter/add_office.rb]: What is the variable user at this stage?

File: lib/commands/crafter/add_office.rb [hunk 2]
```diff
@@ -21,41 +21,46 @@ def prepare(data)
...
-        "The offices available are: #{Office.locations.map(&:capitalize).join(", ")}"
-      else
-        ""
-      end
+    def office_unavailable
+      Logger.info("#{@crafter.user_name} used an unavailable office: #{@office.location}")
+      "The offices available are: #{Office.locations.map(&:capitalize).join(", ")}"
+    end
+
+    def updated_user
```
Comment (fabientownsend) [inline lib/commands/crafter/add_office.rb]: I would delete this method as 100% should be the case now

File: lib/commands/crafter/add_office.rb [hunk 3]
```diff
@@ -21,41 +21,46 @@ def prepare(data)
...
+      "#{user.class.with_office.count}/#{user.class.all.count}"
+    end
+
+    def parse_office_location(data)
```
Comment (fabientownsend) [inline lib/commands/crafter/add_office.rb]: Yeah, I went through quite a few of those. At some point, I think that should be part of where we are reformating the data we receive from Slack, but not part of the commands

File: lib/commands/crafter/add_office.rb (hunk 2 above)
Comment (laurentqro) [reply lib/commands/crafter/add_office.rb]: ok

File: lib/commands/crafter/add_office.rb (hunk 1 above)
Comment (laurentqro) [reply lib/commands/crafter/add_office.rb]: it's a method call to https://github.com/fabientownsend/lunchbot/blob/bfda70732d410480fb78cf19c2d32939fc2d217d/lib/commands/crafter/add_office.rb#L43

```
    def user
      @crafter || @apprentice
end
```

File: lib/commands/crafter/add_office.rb (hunk 3 above)
Comment (laurentqro) [reply lib/commands/crafter/add_office.rb]: yeah probably a good idea to have a `CommandSanitizer` or something that will clean up and format the incoming raw commands before they are passed on to the rest of the app.

**Pull request merged**